mlrun/api/utils/background_tasks.py

```diff
@@ -148,6 +152,7 @@ async def background_task_wrapper(self, name: str, function, *args, **kwargs):
                 name, mlrun.api.schemas.BackgroundTaskState.succeeded
             )
 
+    @mlrun.api.utils.helpers.run_on_chief_only
     def _update_background_task(
```

Comment:
I don't think it makes sense to add the decorator on private functions, public ones should be enough (relevant in other places as well)